daemon/monitor.go

```diff
@@ -54,10 +54,7 @@ func (daemon *Daemon) StateChanged(id string, e libcontainerd.StateInfo) error {
 			"exitCode": strconv.Itoa(int(e.ExitCode)),
 		}
 		daemon.LogContainerEventWithAttributes(c, "die", attributes)
-		if err := c.ToDisk(); err != nil {
-			return err
-		}
-		return daemon.postRunProcessing(c, e)
```

Comment:
Yeah, that's why initially it was there.  Unfortunately, because of the way the interaction between the daemon and libcontainerd is handled, this notification is received in parallel with the restart taking place.  `postRunProcessing` needs to have exclusive access to the container's storage, which can't be guaranteed on a restart.  Rather than try to add some arcane synchronization that the APIs didn't really support, we thought it would be better to restrict to the code path where we know the container is offline for sure.